python/graphstorm/dataloading/sampler.py

```diff
@@ -389,3 +392,299 @@ def sample_blocks(self, g, seed_nodes, exclude_eids=None):
             blocks.insert(0, block)
 
         return seed_nodes, output_nodes, blocks
+
+class FileSamplerInterface:
+    r"""File Sampler Interface.
+
+    Parameters:
+    ---------
+    dataset_path : str
+        Path to the data files.
+        Mutually exclusive with :attr:`files`.
+    """
+
+    def __init__(self, dataset_path):
+        self.files = None
+
+        if not dataset_path:
+            raise ValueError(
+                "Dataset_path={}, should be specified.".format(
+                    dataset_path
+                )
+            )
+        else:
+            if not isinstance(dataset_path, str):
+                raise TypeError(
+                    f"dataset_path should be a str, but got " "dataset_path={dataset_path}"
+                )
+
+            self.dataset_path = dataset_path
+            self.files = [
+                os.path.join(dataset_path, f)
+                for f in os.listdir(dataset_path)
+                if os.path.isfile(os.path.join(dataset_path, f))
+            ]
+            self.files = [f for f in self.files if os.path.getsize(f)>0]
+            self.files.sort()
+
+            if len(self.files)==0:
+                raise ValueError (
+                    f"no non-empty files found at top directory {dataset_path}.")
+
+        self.num_files = len(self.files)
+        logging.info(f"found {self.num_files} files from {dataset_path}")
+
+    def __len__(self):
+        """ Return number of files in the sampler."""
+        return self.num_files
+
+    def __iter__(self):
+        """ Return the iterator of the sampler."""
+        raise NotImplementedError
+
+    def __next__(self):
+        """ Return next file name from the sampler iterator."""
+        raise NotImplementedError
+
+class SequentialFileSampler():
+    r"""Sequential File Sampler. Samples file sequentially.
+
+    Parameters:
+    ----------
+    file_indices : list of int
+        File indices for a local trainer
+    is_train : bool
+        Set to ``True`` if it's training set.
+    """
+    def __init__(self, file_indices, is_train=True):
+        self.file_indices = file_indices
+        self.num_local_files = len(self.file_indices)
+        self._indices = list(range(self.num_local_files))
+        self.is_train = is_train
+
+    def __iter__(self):
+        self._index = -1
+        return self
+
+    def __next__(self):
+        """ Get index for next file"""
+        self._index += 1
+
+        if self._index >= self.num_local_files:
+            if self.is_train:
+                self._index = -1
+            else:
+                # non-infinite sampler if it's for evaluation.
+                # set _index to -1 for next evaluation.
+                self._index = -1
+                return None
+
+        return self.file_indices[self._index]
+
+class RandomShuffleFileSampler():
```

Comment:
The only difference between RandomShuffleFileSampler and SequentialFileSampler is calling rundom.shuffle. Can we combine them?